src/lib/urlPatterns.ts

```diff
@@ -13,6 +14,24 @@ export function extractYouTubeVideoId(url: string) {
   return searchParams.get('v')
 }
 
+export function extractYouTubePlaylistParams(url: string) {
+  const urlObj = new URL(url)
+  const searchParams = new URLSearchParams(urlObj.search)
+  return {
+    list: searchParams.get('list'),
+    index: searchParams.get('index'),
+  }
+}
+
+export const getYoutubeVideoSrc = (url: string) => {
+  const videoId = extractYouTubeVideoId(url)
+  const { list, index } = extractYouTubePlaylistParams(url)
+
+  return `https://www.youtube-nocookie.com/embed/${videoId ? videoId : 'videoseries'}${list ? `?list=${list}` : ''}${
+    !videoId ? `&index=${index ?? 1}` : ''
+  }`
```

Comment:
These ternaries are a bit confusing to follow. It might be better to use the `URL` interface. What do you think?

```suggestion
  const urlObj = new URL(`https://www.youtube-nocookie.com/embed/${videoId ? videoId : 'videoseries'}`)
  if (!videoId) {
    urlObj.searchParams.set('index', index ?? 1)
  }
  if (list) {
    urlObj.searmParams.set('list', list)
  }
  return urlObj.toString()
```